elide-async/src/test/java/com/yahoo/elide/async/service/DefaultResultStorageEngineTest.java

```diff
@@ -0,0 +1,110 @@
+import com.yahoo.elide.ElideSettings;
+import com.yahoo.elide.ElideSettingsBuilder;
+import com.yahoo.elide.async.models.AsyncQuery;
+import com.yahoo.elide.async.models.AsyncQueryResult;
+import com.yahoo.elide.async.models.AsyncQueryResultStorage;
+import com.yahoo.elide.core.DataStore;
+import com.yahoo.elide.core.DataStoreTransaction;
+import com.yahoo.elide.core.EntityDictionary;
+import com.yahoo.elide.core.filter.dialect.RSQLFilterDialect;
+import com.yahoo.elide.security.checks.Check;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.net.URL;
+import java.sql.Blob;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TimeZone;
+
+import javax.sql.rowset.serial.SerialBlob;
+
+public class DefaultResultStorageEngineTest {
+
+    private com.yahoo.elide.async.service.DefaultAsyncQueryDAO asyncQueryDAO;
+    private Elide elide;
+    private DataStore dataStore;
+    private AsyncQuery asyncQuery;
+    private DataStoreTransaction tx;
+    private EntityDictionary dictionary;
+    private com.yahoo.elide.async.service.DefaultResultStorageEngine defaultResultStorageEngine;
+    private AsyncQueryResultStorage asyncQueryResultStorage;
+
+    @BeforeEach
+    public void setupMocks() {
+        dataStore = mock(DataStore.class);
+        asyncQuery = mock(AsyncQuery.class);
+        tx = mock(DataStoreTransaction.class);
+        asyncQueryResultStorage = mock(AsyncQueryResultStorage.class);
+
+        Map<String, Class<? extends Check>> checkMappings = new HashMap<>();
+
+        dictionary = new EntityDictionary(checkMappings);
+        dictionary.bindEntity(AsyncQuery.class);
+        dictionary.bindEntity(AsyncQueryResult.class);
+
+        ElideSettings elideSettings = new ElideSettingsBuilder(dataStore)
+                .withEntityDictionary(dictionary)
+                .withJoinFilterDialect(new RSQLFilterDialect(dictionary))
+                .withSubqueryFilterDialect(new RSQLFilterDialect(dictionary))
+                .withISO8601Dates("yyyy-MM-dd'T'HH:mm'Z'", TimeZone.getTimeZone("UTC"))
+                .build();
+
+        elide = new Elide(elideSettings);
+
+        when(dataStore.beginTransaction()).thenReturn(tx);
+
+        asyncQueryDAO = new com.yahoo.elide.async.service.DefaultAsyncQueryDAO(elide, dataStore);
+        defaultResultStorageEngine = new com.yahoo.elide.async.service.DefaultResultStorageEngine(elide, dataStore, "http://localhost:8080");
+    }
+
+    @Test
+    public void testStoreResults() {
+        String responseBody = "responseBody";
+        byte[] testResponse = responseBody.getBytes();
+        Blob result = null;
+
+        try {
+            result = new SerialBlob(testResponse);
+        } catch (SQLException throwables) {
+            throwables.printStackTrace();
+        }
+
+        URL url = defaultResultStorageEngine.storeResults(
+                asyncQueryResultStorage.getId(), testResponse);
```

Comment:
shouldn't there be a when method for this? asyncQueryResultStorage is a mock object so get will return null.